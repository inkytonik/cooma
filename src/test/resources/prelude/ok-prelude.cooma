{
    type Foo = Int
    def add(l : Foo, r : Foo) Foo = prim IntAdd(l, r)
    val sub = fun(l : Int, r : Int) prim IntSub(l, r)
    val x = {
        a = add(1, 2),
        b = sub(2, 1),
        c = "Hello",
        d = <<e = 3>>
    }
    {}
}
