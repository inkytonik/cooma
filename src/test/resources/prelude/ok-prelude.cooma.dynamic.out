%letc $k1 Foo =
  %letf
    %def add $k2 l =
      %letv $f3 =
        %fun $k4 r =
          %letv $r5 =
            %prim IntAdd l r
          %in $k4 $r5
      %in $k2 $f3
  %in %letc $k6 sub =
    %letc $k7 x =
      %letv $u8 =
        {
        }
      %in %halt $u8
    %in %letv $i14 =
      1
    %in %letc $k12 $r13 =
      %letv $i15 =
        2
      %in %letc $k10 $r11 =
        %letv $i20 =
          2
        %in %letc $k18 $r19 =
          %letv $i21 =
            1
          %in %letc $k16 $r17 =
            %letv $s22 =
              "Hello"
            %in %letv $i24 =
              3
            %in %letv $r23 =
              <<
                e = $i24
              >>
            %in %letv $r9 =
              {
                a = $r11
                b = $r17
                c = $s22
                d = $r23
              }
            %in $k7 $r9
          %in $r19 $k16 $i21
        %in sub $k18 $i20
      %in $r13 $k10 $i15
    %in add $k12 $i14
  %in %letv $f25 =
    %fun $k26 l =
      %letv $f27 =
        %fun $k28 r =
          %letv $r29 =
            %prim IntSub l r
          %in $k28 $r29
      %in $k26 $f27
  %in $k6 $f25
%in %letv $u30 =
  {
  }
%in $k1 $u30
